tester/runner_test.go

```diff
@@ -16,6 +16,67 @@ import (
 	"github.com/open-policy-agent/opa/util/test"
 )
 
+func TestRunner_EnableFailureLine(t *testing.T) {
+
+	ctx := context.Background()
+
+	files := map[string]string{
+		"/a_test.rego": `package foo
+			test_a { 
+				true
+				false
+				true
+			}
+			test_b { 
+				false
+				true
+			}`,
+	}
+
+	tests := map[[2]string]struct {
+		wantErr  bool
+		wantFail bool
+	}{
+		{"data.foo", "test_a"}: {false, true},
+		{"data.foo", "test_b"}: {false, true},
+	}
+
+	test.WithTempFS(files, func(d string) {
+		paths := []string{d}
+		modules, store, err := tester.Load(paths, nil)
+		if err != nil {
+			t.Fatal(err)
+		}
+		ch, err := tester.NewRunner().EnableFailureLine(true).SetStore(store).Run(ctx, modules)
+		if err != nil {
+			t.Fatal(err)
+		}
+		var rs []*tester.Result
+		for r := range ch {
+			rs = append(rs, r)
+		}
+		seen := map[[2]string]struct{}{}
+		for i := range rs {
+			k := [2]string{rs[i].Package, rs[i].Name}
+			seen[k] = struct{}{}
+			exp, ok := tests[k]
+			if !ok {
+				t.Errorf("Unexpected result for %v", k)
+			} else if exp.wantErr != (rs[i].Error != nil) || exp.wantFail != rs[i].Fail {
+				t.Errorf("Expected %v for %v but got: %v", exp, k, rs[i])
+			} else if rs[i].FailedAt == nil {
```

Comment:
❓ I might be missing something, but wouldn't it make sense to assert _what `FailedAt` is_, here?

Now, the next line will always say `Expected Failed Line but got: nil` 😉

Reply:
Do you mean assert the type of structure such as ast.expr?

Reply:
I expected the test to assert the _line of failure_, not just that _there is a line of failure_.

Reply:
hummm, then I would need to hardcode those lines into the test since the line changes for different tests. I could make the failure line the same for all tests and hardcoded it.

Reply:
Yep, that works, submitted new patch.